Enable sklearnex documentation (#587)
intel/scikit-learn-intelex: doc/sources/algorithms.rst

####################

> Supported algorithms

Reviewer: There is no information that just DBSCAN, KMeans, LinReg and LogReg algorithms are supported on GPU for scikit-learn now

Reviewer 2: I think here your help is very much needed in filling out such information, if now we want to see it.

Reviewer 3: There is a larger question - how we would like to present PGU/CPU supported cases? 

Adding new column - might be not sufficient as we will have different params/functionality/data support for CPU and GPU workloads.

Adding new table for GPU? In this case we will have some data duplication.  Are we ok with this?

Author: I think such table will look great in the **GPU support** tab

Author: @napetrov, can we merge this PR and add GPU-supported algorithms in another PR?

Reviewer 3: yes

   * - Clustering
     - KMeans
     - All parameters except ``precompute_distances`` and ``sample_weight`` != None.
     - No limitations.
   * - Clustering
     - DBSCAN
     - All parameters except ``metric`` != 'euclidean' or ``minkowski`` with ``p`` = 2.
     - Only dense data is supported.
   * - Dimensionality reduction
     - PCA
     - All parameters except ``svd_solver`` != 'full'.
     - Sparse data is not supported.
   * - Unsupervised
     - NearestNeighbors
     - All parameters except ``metric`` != 'euclidean' or ``minkowski`` with ``p`` = 2.
     - Sparse data is not supported.
   * - Other
     - train_test_split
     - All parameters are supported.
     - Only dense data is supported.
   * - Other
     - assert_all_finite
     - All parameters are supported.
     - Only dense data is supported.
   * - Other
     - pairwise_distance
     - With metric=``cosine`` and ``correlation``.
     - Only dense data is supported.

Monkey-patched scikit-learn classes and functions passes scikit-learn's own test
suite, with few exceptions, specified in `deselected_tests.yaml
<https://github.com/intel/scikit-learn-intelex/blob/master/deselected_tests.yaml>`__.

> In particular the tests execute `check_estimator

Reviewer 2: Instead, I would add this

Running full the latest scikit-learn test suite with Intel(R) Extension for Scikit-learn: [![CircleCI](https://circleci.com/gh/intel/scikit-learn-intelex.svg?style=svg)](https://circleci.com/gh/intel/scikit-learn-intelex)

   * - Regression
     - Lasso
     - All parameters except ``sample_weight`` != None.
     - Multi-output and sparse data is not supported, #observations should be >= #features.
   * - Clustering
     - KMeans
     - All parameters except ``precompute_distances`` and ``sample_weight`` != None.
     - No limitations.
   * - Clustering
     - DBSCAN
     - All parameters except ``metric`` != 'euclidean' or ``minkowski`` with ``p`` = 2.
     - Only dense data is supported.
   * - Dimensionality reduction
     - PCA
     - All parameters except ``svd_solver`` != 'full'.
     - Sparse data is not supported.
   * - Unsupervised
     - NearestNeighbors
     - All parameters except ``metric`` != 'euclidean' or ``minkowski`` with ``p`` = 2.
     - Sparse data is not supported.
   * - Other
     - train_test_split
     - All parameters are supported.
     - Only dense data is supported.
   * - Other
     - assert_all_finite
     - All parameters are supported.
     - Only dense data is supported.
   * - Other
     - pairwise_distance
     - With metric=``cosine`` and ``correlation``.
     - Only dense data is supported.
   * - Other
     - roc_auc_score

>      - Parameters `average`, `sample_weight`, `max_fpr` and `multi_class` are not supported..

Reviewer 4: ```suggestion
     - Parameters ``average``, ``sample_weight``, ``max_fpr``, and ``multi_class`` are not supported.
```

>      - Parameters ``average``, ``sample_weight``, ``max_fpr`` and ``multi_class`` are not supported..

Reviewer 4: ```suggestion
     - Parameters ``average``, ``sample_weight``, ``max_fpr``, and ``multi_class`` are not supported.
```

     - No limitations.

Monkey-patched scikit-learn classes and functions passes scikit-learn's own test
suite, with few exceptions, specified in `deselected_tests.yaml
<https://github.com/intel/scikit-learn-intelex/blob/master/deselected_tests.yaml>`__.

> Running full the latest scikit-learn test suite with Intel(R) Extension for Scikit-learn*: `**CircleCI**

Reviewer 4: `Running full` -- unclear

Author: I changed this

> Full latest Scikit-learn test suite results with Intel(R) Extension for Scikit-learn*: `CircleCI

Reviewer 4: ```suggestion
The results of the entire latest scikit-learn test suite with Intel(R) Extension for Scikit-learn*: `CircleCI
```
